packages/roosterjs-editor-plugins/lib/plugins/TableResize/preprocessTable.ts

```diff
@@ -0,0 +1,98 @@
+import { normalizeRect } from 'roosterjs-editor-dom';
+
+export const enum Dimension {
+    HEIGHT,
+    WIDTH,
+    BOTH,
+}
+
+export const enum ResizeState {
+    None,
+    Horizontal,
+    Vertical,
+    Both, // when resizing the whole table
+}
+
+export function setHTMLElementSizeInPx(element: HTMLElement, dimension: Dimension) {
+    if (!!element) {
+        if (dimension === Dimension.WIDTH || dimension === Dimension.BOTH) {
+            element.removeAttribute('width');
+            element.removeAttribute('height');
+            const unit = element.style?.width?.substr(element.style.width.length - 2);
+            if (!unit || unit !== 'px' /*&& unit !== 'pt'*/) {
+                const rect = normalizeRect(element.getBoundingClientRect());
+                element.style.boxSizing = 'border-box';
+                element.style.width = `${rect.right - rect.left}px`;
+            }
+        }
```

Comment:
I still think we should not remove width and height that early.

Can we just always get the size using getBoundingClientRect() ? Then after that we apply the calcualted size then we can remove original width/height attribute.

Removing width/height before getBoundingClientRect() may cause the table size changed.

Reply:
Then we don't ned to care the difference of unit (pt, px, ...)

Reply:
can you try again, I set the height to ensure the height doesn't change. There is still minor shifting horizontally but it is only for the first time for Excel. It would be difficult to make it perfect and the change would be expensive. Even in Word, I am seeing similar shifting.

Reply:
Can we just always get the size using getBoundingClientRect() ?

That's the original approach I tried and it would cause the resizing not proportionally, We need to set style.width before reiszing.

Reply:
I believe the size shifit is caused by "box-sizing: border-box". When set this style, border width will be counted as part of TD width. so that when we add this style if it was not there, we also need to calculate the width/size with border width.

By saying using getBoundingClientRect(), I mean:

1. When start to resize, calculate size of each cell using getBoundingClientRect.
2. After get size of all cells, write them back in CSS width/height, and remove HTML attribute width/height
3. Then we can really start to resize.

So that is still "before resizing"

The difference is, we never read the original HTML width/height and CSS width/height. Instead, we read the value from getBoundingClientRect(). Because the real size can be different than width/height in HTML or CSS when there is size conflict between TABLE/TR/TD width/height. Browser will calculate a size for us which may be different than the value in HTML.